kloppy/domain/services/event_deducers/carry.py

```diff
@@ -0,0 +1,142 @@
+            if event.event_type not in valid_event_types:
+                continue
+            idx_plus = 1
+            generic_next_event = True
+            while idx + idx_plus < len(dataset.events) and generic_next_event:
+                next_event = dataset.events[idx + idx_plus]
+
+                if next_event.event_type in [
+                    EventType.GENERIC,
+                    EventType.PRESSURE,
+                ]:
+                    idx_plus += 1
+                    continue
+                else:
+                    generic_next_event = False
+                if not event.team.team_id == next_event.team.team_id:
+                    continue
+
+                if next_event.event_type not in valid_event_types:
+                    continue
+                # not headed shot
+                if (
+                    (hasattr(next_event, "body_part"))
+                    and (next_event.event_type == EventType.SHOT)
+                    and (
+                        next_event.body_part.type.isin(
+                            [BodyPart.HEAD, BodyPart.HEAD_OTHER]
+                        )
+                    )
+                ):
+                    continue
+
+                if hasattr(event, "end_coordinates"):
+                    last_coord = event.end_coordinates
+                elif hasattr(event, "receiver_coordinates"):
+                    last_coord = event.receiver_coordinates
+                else:
+                    last_coord = event.coordinates
+
+                new_coord = next_event.coordinates
+
+                distance_meters = pitch.distance_between(
+                    new_coord, last_coord, Unit.METERS
+                )
+                # Not far enough
+                if distance_meters < self.min_carry_length_meters:
+                    continue
+                # Too far
+                if distance_meters > self.max_carry_length_meters:
+                    continue
+
+                dt = next_event.timestamp - event.timestamp
+                # not same phase
+                if dt > self.max_carry_duration:
+                    continue
+                # not same period
+                if not event.period.id == next_event.period.id:
+                    continue
+
+                if hasattr(event, "end_timestamp"):
+                    last_timestamp = event.end_timestamp + timedelta(
+                        seconds=0.1
+                    )
+                elif hasattr(event, "receive_timestamp"):
+                    last_timestamp = event.receive_timestamp + timedelta(
+                        seconds=0.1
+                    )
+                else:
+                    last_timestamp = (
+                        event.timestamp
+                        + (next_event.timestamp - event.timestamp) / 10
+                    )
+
+                generic_event_args = {
+                    "event_id": f"{str(uuid.uuid4())}",
```

Comment:
We've adopted the convention of adding the prefix "synthetic-" for generated events.